packages/web-components/fast-foundation/src/number-field/number-field.ts

```diff
@@ -242,22 +242,29 @@ export class NumberField extends FormAssociatedNumberField {
         super.validate(this.control);
     }
 
+    private limitPrecision(value: number): number {
+        return parseFloat(value.toPrecision(15));
+    }
+
     /**
      * Sets the internal value to a valid number between the min and max properties
      * @param value - user input
      *
      * @internal
      */
     private getValidValue(value: string): string {
-        let validValue: number | string = parseFloat(parseFloat(value).toPrecision(15));
+        let validValue: number | string = this.limitPrecision(parseFloat(value));
```

Comment:
Unless my mental type propagation is missing something, this local value is never a string anymore.
```suggestion
        let validValue: number = this.limitPrecision(parseFloat(value));
```